django/test/selenium.py

```diff
@@ -214,7 +216,9 @@ def high_contrast(self):
     def take_screenshot(self, name):
         if not self.screenshots:
             return
-        path = Path.cwd() / "screenshots" / f"{self._testMethodName}-{name}.png"
+        test = getattr(self, self._testMethodName)
+        filename = f"{test._screenshot_name}-{name}.png"
+        path = Path.cwd() / "screenshots" / test._screenshot_case / filename
```

Comment:
For context, most of the time people update 1 selenium test and they need to be reminded about the other "cases" and so I think having all of those cases together is nice
I also like to go through all of the images and check nothing unexpected is popping up from some css changes

I'm still not sure what the best way to organise them would be (having this by admin section could be nice e.g. "change_list", or we can mimic the current test structure folder). 

I'm tempted to try and push greater adoption across the test suite and people "use them" as part of their process before we make a decision to organise it. What are your thoughts here?

Reply:
> I'm still not sure what the best way to organise them would be (having this by admin section could be nice e.g. "change_list", or we can mimic the current test structure folder).

I'm concerned that would just become a soup of too many folders, each with very few images in. Having said that, currently we're only using the test method name which could clash...

I guess there are not too many images as yet, so maybe a slightly different re-organization of name would be fine for now to better group things. Perhaps `<test-name>--<name>--<case>.png` instead of `<test-name>_<case>-<name>.png`? (Also, using the double dash will make it easier to see as the name often has hyphens in it.)

<details>
<summary>The status quo...</summary>

```console
$ tree tests/screenshots/
tests/screenshots/
├── test_collapsible_fieldset_dark-collapsed.png
├── test_collapsible_fieldset_dark-expanded.png
├── test_collapsible_fieldset_desktop_size-collapsed.png
├── test_collapsible_fieldset_desktop_size-expanded.png
├── test_collapsible_fieldset_mobile_size-collapsed.png
├── test_collapsible_fieldset_mobile_size-expanded.png
├── test_first_field_focus_dark-focus-multi-widget.png
├── test_first_field_focus_dark-focus-single-widget.png
├── test_first_field_focus_desktop_size-focus-multi-widget.png
├── test_first_field_focus_desktop_size-focus-single-widget.png
├── test_first_field_focus_mobile_size-focus-multi-widget.png
├── test_first_field_focus_mobile_size-focus-single-widget.png
├── test_first_field_focus_rtl-focus-multi-widget.png
├── test_first_field_focus_rtl-focus-single-widget.png
├── test_login_button_centered_dark-login.png
├── test_login_button_centered_desktop_size-login.png
├── test_login_button_centered_mobile_size-login.png
├── test_login_button_centered_rtl-login.png
├── test_selectbox_height_collapsible_fieldset_dark-selectbox-collapsible.png
├── test_selectbox_height_collapsible_fieldset_desktop_size-selectbox-collapsible.png
├── test_selectbox_height_collapsible_fieldset_mobile_size-selectbox-collapsible.png
├── test_selectbox_height_collapsible_fieldset_rtl-selectbox-collapsible.png
├── test_selectbox_height_not_collapsible_fieldset_dark-selectbox-non-collapsible.png
├── test_selectbox_height_not_collapsible_fieldset_desktop_size-selectbox-non-collapsible.png
├── test_selectbox_height_not_collapsible_fieldset_mobile_size-selectbox-non-collapsible.png
└── test_selectbox_height_not_collapsible_fieldset_rtl-selectbox-non-collapsible.png
```

</details>

<details>
<summary>My original proposal...</summary>

```console
$ tree tests/screenshots/
tests/screenshots/
├── dark
│   ├── test_collapsible_fieldset-collapsed.png
│   ├── test_collapsible_fieldset-expanded.png
│   ├── test_first_field_focus-focus-multi-widget.png
│   ├── test_first_field_focus-focus-single-widget.png
│   ├── test_login_button_centered-login.png
│   ├── test_selectbox_height_collapsible_fieldset-selectbox-collapsible.png
│   └── test_selectbox_height_not_collapsible_fieldset-selectbox-non-collapsible.png
├── desktop_size
│   ├── test_collapsible_fieldset-collapsed.png
│   ├── test_collapsible_fieldset-expanded.png
│   ├── test_first_field_focus-focus-multi-widget.png
│   ├── test_first_field_focus-focus-single-widget.png
│   ├── test_login_button_centered-login.png
│   ├── test_selectbox_height_collapsible_fieldset-selectbox-collapsible.png
│   └── test_selectbox_height_not_collapsible_fieldset-selectbox-non-collapsible.png
├── mobile_size
│   ├── test_collapsible_fieldset-collapsed.png
│   ├── test_collapsible_fieldset-expanded.png
│   ├── test_first_field_focus-focus-multi-widget.png
│   ├── test_first_field_focus-focus-single-widget.png
│   ├── test_login_button_centered-login.png
│   ├── test_selectbox_height_collapsible_fieldset-selectbox-collapsible.png
│   └── test_selectbox_height_not_collapsible_fieldset-selectbox-non-collapsible.png
└── rtl
    ├── test_first_field_focus-focus-multi-widget.png
    ├── test_first_field_focus-focus-single-widget.png
    ├── test_login_button_centered-login.png
    ├── test_selectbox_height_collapsible_fieldset-selectbox-collapsible.png
    └── test_selectbox_height_not_collapsible_fieldset-selectbox-non-collapsible.png
```
</details>

<details>
<summary>My updated proposal...</summary>

```console
$ tree tests/screenshots/
tests/screenshots/
├── test_collapsible_fieldset--collapsed--dark.png
├── test_collapsible_fieldset--collapsed--desktop_size.png
├── test_collapsible_fieldset--collapsed--mobile_size.png
├── test_collapsible_fieldset--expanded--dark.png
├── test_collapsible_fieldset--expanded--desktop_size.png
├── test_collapsible_fieldset--expanded--mobile_size.png
├── test_first_field_focus--focus-multi-widget--dark.png
├── test_first_field_focus--focus-multi-widget--desktop_size.png
├── test_first_field_focus--focus-multi-widget--mobile_size.png
├── test_first_field_focus--focus-multi-widget--rtl.png
├── test_first_field_focus--focus-single-widget--dark.png
├── test_first_field_focus--focus-single-widget--desktop_size.png
├── test_first_field_focus--focus-single-widget--mobile_size.png
├── test_first_field_focus--focus-single-widget--rtl.png
├── test_login_button_centered--login--dark.png
├── test_login_button_centered--login--desktop_size.png
├── test_login_button_centered--login--mobile_size.png
├── test_login_button_centered--login--rtl.png
├── test_selectbox_height_collapsible_fieldset--selectbox-collapsible--dark.png
├── test_selectbox_height_collapsible_fieldset--selectbox-collapsible--desktop_size.png
├── test_selectbox_height_collapsible_fieldset--selectbox-collapsible--mobile_size.png
├── test_selectbox_height_collapsible_fieldset--selectbox-collapsible--rtl.png
├── test_selectbox_height_not_collapsible_fieldset--selectbox-non-collapsible--dark.png
├── test_selectbox_height_not_collapsible_fieldset--selectbox-non-collapsible--desktop_size.png
├── test_selectbox_height_not_collapsible_fieldset--selectbox-non-collapsible--mobile_size.png
└── test_selectbox_height_not_collapsible_fieldset--selectbox-non-collapsible--rtl.png
```

</details>

(I'll have to look into this more later.)

Reply:
Yes! I think that's nicer 👍

Reply:
I've updated to `<test-name>--<name>--<case>.png`, but still need to consider perhaps making use of the `__module__` and `__qualname__`.